src/Avalonia.Animation/Animator`1.cs

```diff
@@ -28,21 +28,20 @@ public abstract class Animator<T> : AvaloniaList<AnimatorKeyFrame>, IAnimator
         public Animator()
         {
             // Invalidate keyframes when changed.
-             this.CollectionChanged += delegate { _isVerifiedAndConverted = false; };
+            this.CollectionChanged += delegate { _isVerifiedAndConverted = false; };
         }
 
         /// <inheritdoc/>
         public virtual IDisposable Apply(Animation animation, Animatable control, IObservable<bool> match, Action onComplete)
         {
-             if (!_isVerifiedAndConverted)
+            if (!_isVerifiedAndConverted)
                 VerifyConvertKeyFrames();
 
-            return match
-                .Where(p => p)
-                .Subscribe(_ =>
-                {
-                    var timerObs = RunKeyFrames(animation, control, onComplete);
-                });
+            return match.DistinctUntilChanged()
+                        .Select(x => x ? RunKeyFrames(animation, control, onComplete) : null)
+                        .Buffer(2, 1)
```

Comment:
Are we sure that `.Buffer(2, 1)` does what we want? Based on the [System.Reactive documentation](http://reactivex.io/documentation/operators/buffer.html) it doesn't seem right. I think we might need more of a custom observable similar to https://github.com/jkoritzinsky/AvaloniaAV/blob/master/AvaloniaAV/ObservableExtensions.cs but based off `LightweightObservableBase` to actually do what we want.